pkg/network/port_test.go

```diff
@@ -131,11 +132,27 @@ func testReadInitialState(t *testing.T, proto string) {
 
 	var ports []uint16
 	for _, proto := range protos {
-		ports = append(ports, runServerProcess(t, proto, 0, rootNs))
+		i := 0
+		for ; i < 5; i++ {
+			p := runServerProcess(t, proto, 0, rootNs)
+			if !slices.Contains(ports, p) {
+				ports = append(ports, runServerProcess(t, proto, 0, rootNs))
```

Comment:
you're entering this condition if the port `p` does not exists in the list `ports`, but then you add to the list the value returned from `runServerProcess(t, proto, 0, rootNs)` (maybe you either meant `p`, or `runServerProcess(t, proto, p, rootNs)`), which returns a new port (since you passed `0` in the 3rd argument

Reply:
:face_palm: